pkg/util/system/system_utils.go

```diff
@@ -17,11 +17,20 @@ limitations under the License.
 package system
 
 import (
-	"regexp"
+	"strings"
 )
 
 // TODO: find a better way of figuring out if given node is a registered master.
 func IsMasterNode(nodeName string) bool {
-	r := regexp.MustCompile("master(-...)?$")
```

Comment:
i think most of the allocation are coming from the compile phase. we can compile once instead of compiling the regex every time. If that can save enough allocation, we can get rid of all the comments here.